spec/models/factory_log_spec.rb

```diff
@@ -1,97 +1,29 @@
-    let!(:factoryAs) { create_list(:factory_log, 10) }
-    let!(:factoryAs_short) { create_list(:factory_log, 7, time: 0.5) }
-    let!(:factoryAs_long) { create_list(:factory_log, 3, time: 5.0) }
-    let(:sample) { factoryAs.first }
-
-    it 'return average time factoryAs' do
-      expect(FactoryLog.average_time(sample.name + sample.traits)).to eq ((2.0 * 10) + (0.5 * 7) + (5.0 * 3)) / (10 + 7 + 3)
-    end
-  end
-
-  describe 'time_ranking' do
-    let!(:factoryAs) { create_list(:factory_log, 10) }
-    let!(:factoryAs_with_traits) { create_list(:factory_log, 1, :have_trait) }
-    let!(:factoryBs) { create_list(:factory_log, 3, :factoryB) }
-
-    let!(:a_sample) { factoryAs.first }
-    let!(:a_with_traits_sample) { factoryAs_with_traits.first }
-    let!(:b_sample) { factoryBs.first }
-
-    let!(:a_ranking_data) {
-                            {
-                              name: (a_sample.name + a_sample.traits),
-                              total_time: 2.0 * 10,
-                              count: 10,
-                              average_time: 2.0
-                            }
-                          }
-    let!(:a_with_traits_ranking_data) {
-                                        {
-                                          name: (a_with_traits_sample.name + a_with_traits_sample.traits),
-                                          total_time: 2.0,
-                                          count: 1,
-                                          average_time: 2.0
-                                        }
-                                      }
-    let!(:b_ranking_data) {
-                            {
-                              name: (b_sample.name + b_sample.traits),
-                              total_time: 10.0 * 3,
-                              count: 3,
-                              average_time: 10.0
-                            }
-                          }
-
-    it 'ranking about time they took (total)' do
-      expect(FactoryLog.time_ranking).to eq [
-                                              b_ranking_data,
-                                              a_ranking_data,
-                                              a_with_traits_ranking_data
-                                            ]
+  describe "time ranking" do
+    let!(:inspected_factory1) { { name: "factory1", traits: [], assos: [] } }
+    let!(:inspected_factory2) { { name: "factory1", traits: ["trait_a"], assos: [] } }
+    let!(:factory1) { Factory.create_unique_factory(inspected_factory1) }
+    let!(:factory2) { Factory.create_unique_factory(inspected_factory2) }
+    it do
+      FactoryLog.logging(inspected_factory1, 1.0)
+      FactoryLog.logging(inspected_factory1, 1.1)
+      FactoryLog.logging(inspected_factory1, 1.0)
+      FactoryLog.logging(inspected_factory2, 2.1)
+      expect(FactoryLog.ranking).to eq [
+                                         {
+                                            factory: factory1,
```

Comment:
Style/IndentHash: Use 2 spaces for indentation in a hash, relative to the start of the line where the left curly brace is.